### DISCUSSION TITLE
How to add Plot.plot() to container?

### DISCUSSION THREAD

**Discussion by Martien** [observablehq/plot#1290] [Q&A] [answered]
Suppose I have a container (used to hold a plot, some controls, and show some values, like an instrument panel):

`const panel = d3.create("svg").attr("width", width).attr("height", width * 9 / 16);`

How do I append a `Plot.plot()` to that container?

**Comment by Fil** [comment, marked as answer]
If `panel` is a [selection](https://github.com/d3/d3-selection), you can use the _selection_.append method:

~~~js
panel.append(() => Plot.plot(…))
~~~

**Reply by mbostock** [reply]
Note that Plot.plot can sometimes generate an HTML figure element (_e.g._, when you supply a caption or ask for a color legend), so you should be careful if you go inserting the result into another SVG element.

**Comment by Martien** [comment]
Thank you @mbostock and @fil and @mootari. This works fine. I missed the `() => Plot.plot(…)` option completely and tripped over the `caption` issue as well. It made me thoroughly reread and grasp [_selection_.append](https://github.com/d3/d3-selection#selection_append), to much delight. It’s all there.

I also noted that the `backgroundColor` set in `Plot.plot(…)` is ignored (at sublevels), so I had to set it at the top level myself.

[I’m confused about getting the original replies from Fabian yesterday, and that they are now from Mike and Fil.]

**Reply by mootari** [reply]
> I’m confused about getting the original replies from Fabian yesterday

I don't follow? I never took part in this discussion.

**Reply by Martien** [reply]
Ah, well, even more confusing, as my inbox shows this:
<img width="1037" alt="Screenshot 2023-02-22 at 13 11 09" src="https://user-images.githubusercontent.com/6566925/220616730-83f1e7ee-e224-478d-bf9f-33eb229212ee.png">

**Reply by Martien** [reply]
I’ll check my Mail and Contacts. Perhaps the confusion is there.

**Comment by Martien** [comment]
I think I got it: <img width="950" alt="Screenshot 2023-02-22 at 13 17 12" src="https://user-images.githubusercontent.com/6566925/220617499-7177ec43-1632-48b8-ac2e-ec26c94b7087.png">
